ddtrace/settings/config.py

```diff
@@ -187,6 +196,72 @@ def get_error_ranges(error_range_str):
     return error_ranges  # type: ignore[return-value]
 
 
+ConfigSource = Literal["default", "env", "code"]
+
+
+class _ConfigItem:
+    """Configuration item that tracks the value of a setting, and where it came from."""
+
+    def __init__(self, name, default, envs):
+        self._name = name
+        self._default_value = default
+        self._env_value = None
+        self._code_value = None
+        self._envs = envs
+        for (env_var, parser) in envs:
+            if env_var in os.environ:
+                self._env_value = parser(os.environ[env_var])
+                break
```

Comment:
This break is useful when one configuration can be set by multiple environment variables. In this scenario the first one set wins. However this behavior is pretty subtle and could be hard to debug. Should we log a warning if multiple env vars are set? 

For example if a customer sets `DD_TRACE_PROPAGATION_STYLE` and `DD_TRACE_PROPAGATION_STYLE_EXTRACT`. `DD_TRACE_PROPAGATION_STYLE_EXTRACT` will take precedence and `DD_TRACE_PROPAGATION_STYLE` will be ignored.

Reply:
Yes, I think this is a good follow-up. We can use the `_ConfigItem` to generate documentation for saying what the environment variable precedence is.